### DISCUSSION TITLE
kratos-courier-0 no endpoint

### DISCUSSION THREAD

**Discussion by williamohara** [ory/kratos#2581] [Q&A] [answered]
the courier service that get deployed through a service set - it has no ports . 
I installed using helm
the yaml `statefulset-mail.yaml` does not have ports defined on the container spec
but
the yaml `statefulset-svc.yaml` has ports defined in its spec..

is there a reason for that?

I am using bridge to kubernetes to bridge my local machine to my cluster so I can do testing. It fails when it comes across a service on my cluster without an endpoint. 

I made my own template files and added container ports to the `statefulset-mail.yaml` file and it no has an endpoint...

but it still does not have a cluster ip...

```
~ $ kubectl get endpoints
NAME                                               ENDPOINTS                        AGE
kratos-admin                                       10.240.1.39:4434                 11m
kratos-courier                                     10.240.1.11:4433                 11m
kratos-public                                      10.240.1.39:4433                 11m
kubernetes                                         20.121.146.165:443               4h31m
newauthnode                                        10.240.0.13:3000                 4h9m
nginx-ingress-ingress-nginx-controller             10.240.0.90:443,10.240.0.90:80   4h24m
nginx-ingress-ingress-nginx-controller-admission   10.240.0.90:8443                 4h24m
~ $ kubectl get services
NAME                                               TYPE           CLUSTER-IP     EXTERNAL-IP    PORT(S)                      AGE
kratos-admin                                       ClusterIP      10.0.164.237   <none>         4434/TCP                     11m
kratos-courier                                     ClusterIP      None           <none>         4433/TCP                     11m
kratos-public                                      ClusterIP      10.0.152.235   <none>         4433/TCP                     11m
kubernetes                                         ClusterIP      10.0.0.1       <none>         443/TCP                      4h31m
newauthnode                                        ClusterIP      10.0.89.122    <none>         3000/TCP                     4h9m
nginx-ingress-ingress-nginx-controller             LoadBalancer   10.0.2.105     20.121.77.93   80:31995/TCP,443:30170/TCP   4h25m
nginx-ingress-ingress-nginx-controller-admission   ClusterIP      10.0.211.197   <none>         443/TCP                      4h25m
```

**Comment by vinckr** [comment]
Hello @williamohara 
please forgive the late answer. 
Is this an issue related to Kratos, the Helm charts, or Kubernetes?

**Reply by williamohara** [reply, marked as answer]
It is actually a bug with Bridge to Kubernetes https://github.com/microsoft/mindaro/issues/288

It is looking for ip addresses on endpoints and when it can't find one - fails. As a workaround I made my own helm template from yours and added a port to spec.template.spec.containers.port in statefulset-mail.yaml and use it to deploy.

i think it is innocuous to do so  but I want to be as tight as possible.
